h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -847,11 +902,81 @@ private GLMModel runGLMModel(boolean standardize, Family family, Link link, Key<
     GLMModel model = new GLM(tempParams).trainModel().get();
     return model;
   }
+  
+  // copy from scoring_history back to _sc or _lsc
+  private void reinstallSC() {
+      TwoDimTable scoringHistory = _model._output._scoring_history;
+      String[] colHeaders2Restore = _parms._lambda_search ? 
+              new String[]{"iteration", "timestamp", "lambda", "predictors", "deviance_train", 
+                      "deviance_test"}
+              : new String[]{"iteration", "timestamp", "negative_log_likelihood", "objective", "sum(etai-eta0)^2", 
+              "convergence"};
+      int num2Copy = _parms._HGLM || _parms._lambda_search ? colHeaders2Restore.length : colHeaders2Restore.length-2;
+      int[] colHeadersIndex = grabHeaderIndex(scoringHistory, num2Copy, colHeaders2Restore);
+      if (_parms._lambda_search)
+        restoreLSC(scoringHistory, colHeadersIndex);
+      else
+        restoreSC(scoringHistory, colHeadersIndex);
+  }
+  
+  private void restoreLSC(TwoDimTable sHist, int[] colIndices) {
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
```

Comment:
Formatter should be created only once, it is defined on multiple places in this file. I would suggest creating static constant for it.

Reply:
Good catch.